src/app/services/catalogue/catalogue.service.ts

```diff
@@ -37,16 +37,44 @@ export class CatalogueService {
     const topicFilter = topic ? `+topic:${topic}` : '';
     const orgFilter = org ? `org:${org}+` : '';
 
-    return this.getFromAPI(
-      `https://api.github.com/search/repositories?page=${this.CONF.page}&per_page=${this.CONF.perPage}&q=${orgFilter}fork:true${topicFilter}+sort:stars`,
-      key,
+    const perPage = this.CONF.perPage;
+    let totalSize = null;
+
+    const getPage = (page) => {
+      return this.http.get<RepositoryPage>(`https://api.github.com/search/repositories?page=${page}&per_page=${this.CONF.perPage}&q=${orgFilter}fork:true${topicFilter}+sort:stars`);
+    };
```

Comment:
```suggestion
    const getPage = (page: string) =>
      this.http.get<RepositoryPage>(
        `https://api.github.com/search/repositories?page=${page}&per_page=${this.CONF.perPage}&q=${orgFilter}fork:true${topicFilter}+sort:stars`,
      );
```

Reply:
Thanks for suggestion, I just changed the type to `number`.